lib/api/user_repo.dart

```diff
@@ -23,21 +23,33 @@ class UserRepositoryImpl implements UserRepository {
     return mockUser();
   }
 
-  @override
-  Future<User> retrieveUserById(String id) async {
-    final QueryOptions options =
-        QueryOptions(documentNode: gql(queryUser(id)));
 
-    final QueryResult queryResult = await utility.client.query(options);
+  static User mockUser() {
+    return new User(name: "test", email: "test@flutter.ca");
+  }
+
+  @override
+  Future<User> updateUserEmail(String id, String name, String newEmail) async {
+    final MutationOptions mutationOptions = MutationOptions(documentNode: gql(UserDocument.updateUserEmail(id, newEmail)));
+    final QueryResult queryResult = await utility.client.mutate(mutationOptions);
     if (queryResult.hasException) {
       log("Error fetching profile details ${queryResult.exception}");
     }
     return User(
-        name: queryResult.data['user']['name'],
-        email: queryResult.data['user']['email']);
+        name: name,
+        email: queryResult.data['updateUser']['user']['email']);
   }
 
-  static User mockUser() {
-    return new User(name: "test", email: "test@flutter.ca");
+  @override
+  Future<User> updateUsersName(String id, String newName, String email) async {
```

Comment:
Should email be returned here? I think we can just update the query to return the full user object and just return a user object parsed from `queryResult.data['updateUser']['user']`

Reply:
Good point